src/basic/env-util.c

```diff
@@ -419,7 +419,7 @@ char *strv_env_get_n(char **l, const char *name, size_t k) {
         if (k <= 0)
                 return NULL;
 
-        STRV_FOREACH(i, l)
+        STRV_FOREACH_BACKWARDS(i, l)
```

Comment:
hmm, so this causes us to traverse through the array twice to get the right value. I'd prefer if we'd only have to do that once, i.e. iterate from front to back, and store the last matching item we found but continue iterating until the end, and then return that.